File: 206-reverse-linked-list/206-reverse-linked-list.java

class Solution {
    public ListNode reverseList(ListNode head) {
        ListNode prev = null;
        ListNode current = head;
        ListNode backup;
        
        while (current != null) {
            backup = current.next; // node with value 2
            current.next = prev; // reversing the pointers
            prev = current; // let's move on
            current = backup; // let's move on
        }
        return prev; // the new head
    }
}

File: 206-reverse-linked-list/206-reverse-linked-list.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} head
 * @return {ListNode}
 */

var reverseList = function(head) {
    let first = null;
    let second = head;
    let backup = null;
    
    while (second != null) {
        backup = second.next;
        second.next = first;
        first = second;
        second = backup;
    }
    return first;
};















































